UI: allow navigation to subdirectory (#22436)
hashicorp/vault: website/content/api-docs/secret/pki.mdx

| `ACME` | `/pki/acme/directory`                                              | `role:role_ref`            | Specified by the role | `:role_ref`   |
| `ACME` | `/pki/acme/directory`                                              | `external-policy(:policy)` | Specified by CIEPS    | CIEPS         |
| `ACME` | `/pki/issuer/:issuer_ref/acme/directory`                           | `sign-verbatim`            | `:issuer_ref`         | Sign-Verbatim |
| `ACME` | `/pki/issuer/:issuer_ref/acme/directory`                           | `role:role_ref`            | `:issuer_ref`         | `:role_ref`   |
| `ACME` | `/pki/roles/:role/acme/directory`                                  | (any)                      | Specified by the role | `:role`       |
| `ACME` | `/pki/issuer/:issuer_ref/roles/:role/acme/directory`               | (any)                      | `:issuer_ref`         | `:role`       |
| `ACME` | `/pki/external-policy(/:policy)/acme/directory`                    | (any)                      | Specified by CIEPS    | CIEPS         |
| `ACME` | `/pki/issuer/:issuer_ref/external-policy(/:policy)/acme/directory` | (any)                      | Specified by CIEPS    | CIEPS         |

> When a role is not explicitly specified, behavior is specified by the
> `default_directory_policy` in the [ACME configuration](#set-acme-configuration).
> These directories can also be forbidden by setting that policy as `forbid`.  If
> the policy is `sign-verbatim` then _any_ identifier for which the client can
> prove ownership of will be issued for.  This is similar to using the
> [Sign Verbatim](#sign-verbatim) endpoint, but with additional verification
> that the client has proven ownership (within the ACME protocol) of the
> requested certificate identifiers. When `external-policy` is specified as the
> default value, the CIEPS engine <EnterpriseAlert inline="true" /> is used for
> validating and templating the certificate instead of a role; ACME's challenge
> validation is still enforced. An optional policy name can be specified by using
> `external-policy:policy`. Roles are not used when CIEPS is used.

Reviewer: ```suggestion
When a role is not explicitly specified, Vault uses the default policy set by
`default_directory_policy` in the [ACME configuration](#set-acme-configuration):
- `forbid`: denies access to the directory
- `sign-verbatim`: allows access to clients that provide *any* identifier
  they can prove ownership for. `sign-verbatim` behaves like the 
  [Sign Verbatim](#sign-verbatim) endpoint but with the additional verification
  that the client can prove ownership within the ACME protocol of the requested
  certificate identifiers.
- `external-policy`: <EnterpriseAlert inline="true" /> uses the CIEPS engine to
  validate and template the certificate instead of a role with ACME challenge
  validation. You can specify an optional policy name with `external-policy:policy`.
  Roles are not used when CIEPS is used.
```

Style correction: write in active voice, make options scannable

}
```

### Generate certificate and key with external policy <EnterpriseAlert inline="true" />

> Similar to the [generate certificate and key](#generate-certificate-and-key)
> endpoint, this endpoint generate key material and certificate via an external
> policy engine. The private key material stays local to Vault, with the external
> service getting only an empty CSR. Any parameters passed to this endpoint are
> passed verbatim to the CIEPS engine <EnterpriseAlert inline="true" />. The
> response format is the same between both endpoints.

Reviewer: ```suggestion
Similar to the [generate certificate and key](#generate-certificate-and-key)
endpoint, this endpoint generate key material and certificate via an external
policy engine. The private key material stays local to Vault, with the external
service getting only an empty CSR. Any parameters passed to this endpoint are
passed verbatim to the CIEPS engine <EnterpriseAlert inline="true" />. The
response format is the same between both endpoints.
``````suggestion
Similar to the [generate certificate and key](#generate-certificate-and-key)
endpoint, the external policy endpoint generates key material and certificate
through an external policy engine. The private key material stays local to
Vault and the external service receives an empty CSR. All parameters passed
to the endpoint are passed, verbatim, to the CIEPS engine <EnterpriseAlert inline="true" />.
The response format is the same between both endpoints.
```

Style correction: simplify language, write in active voice

> It is suggested to limit access to the path-overridden issue endpoint (on
> `/pki/issuer/:issuer_ref/external-policy/issue/:policy`) and let the CIEPS
> engine override the issuer as necessary.

Reviewer: ```suggestion
We recommend you limit access to the path-overridden issue endpoint (on
`/pki/issuer/:issuer_ref/external-policy/issue/:policy`) and let the CIEPS
engine override the issuer as necessary.
```

Style correction: active voice
